Model/TransUnit.php

```diff
@@ -152,7 +152,7 @@ public function hasTranslation($locale)
      * Return the content of translation for the given locale.
      *
      * @param string $locale
-     * @return \Lexik\Bundle\TranslationBundle\Model\Translation
+     * @return \Lexik\Bundle\TranslationBundle\Model\Translation|null
```

Comment:
If you are sure it can be null, please add a nullable return type to the function itself. As we now rely on php 8 features to be in place, this can be safely done imo. We could also conside to have things like these fixed automatically with rector.

Reply:
Updated.

I'm not sure I understand what you mean by "we now rely on php8 features to be in place". We should still support php7.4 according to https://github.com/lexik/LexikTranslationBundle/blob/master/composer.json#L20

Reply:
Yes, you are right. I got a little bit ahead of myself. My intention is to freeze that at branch 6 and then move master forward to php 8.1 and the latest and greatest SF version. I'm not 100% sure where php74 is with nullable return types. Probably works.

Reply:
Sounds like a good plan! And yes, nullable return types work since 7.1 so no worries.